test/suite/test_rollback_to_stable10.py

```diff
@@ -52,9 +52,21 @@ class test_rollback_to_stable10(test_rollback_to_stable_base):
     scenarios = make_scenarios(key_format_values, prepare_values)
 
     def conn_config(self):
-        config = 'cache_size=6MB,statistics=(all),statistics_log=(json,on_close,wait=1),log=(enabled=true),timing_stress_for_test=[history_store_checkpoint_delay]'
+        config = 'cache_size=25MB,statistics=(all),statistics_log=(json,on_close,wait=1),log=(enabled=true),timing_stress_for_test=[history_store_checkpoint_delay]'
         return config
 
+    def evict_cursor(self, uri, nrows, check_value):
+        # Configure debug behavior on a cursor to evict the page positioned on when the reset API is used.
+        evict_cursor = self.session.open_cursor(uri, None, "debug=(release_evict)")
+        self.session.begin_transaction("ignore_prepare=true")
+        for i in range (1, nrows + 1):
+            evict_cursor.set_key(i)
+            self.assertEqual(evict_cursor.search(), 0)
+            self.assertEqual(evict_cursor.get_value(), check_value)
+            evict_cursor.reset()
```

Comment:
No need to reset?

Reply:
Evict cursor only performs eviction after reset?

Reply:
Is it the case that if you don't reset explicitly, it'll evict the page when it moves to the next page, but if you do, it'll evict the page repeatedly, once for each key? This is something I've been wondering about...

Reply:
Currently, it is not guaranteed that a page gets evicted when the cursor moves to the next page. It would be worth implementing that approach for the evict cursors instead of evicting whenever the `reset` is called. That approach avoids the unnecessary page evicts for every key on the page. I can do the reset for every 10 keys can reduce the unnecessary page evicts.